### PR TITLE
Remove UUIDs from being service name of designs and add depends-on

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR adds the following feature-

1. The name of the service in patterfile generated by design would be unique as well as valid and human readable.
2. A node inside of a node-group will have a depends-on relation in the patternfile generated

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Thanks for knocking this out, @Revolyssup

**Comment by leecalcote** [comment]
merge conflict...

File: models/pattern/core/pattern.go [hunk 1]
```diff
@@ -202,23 +204,74 @@ func NewPatternFileFromCytoscapeJSJSON(byt []byte) (Pattern, error) {
...
+		childSvc := eleToSvc[child]
+		if childSvc != "" {
+			for _, parent := range parents {
+				pf.Services[childSvc].DependsOn = append(pf.Services[childSvc].DependsOn, eleToSvc[parent])
+			}
+		}
+	}
+	return pf, err
+}
+func getRandomAlphabetsOfDigit(length int) (s string) {
```
Comment (leecalcote) [inline models/pattern/core/pattern.go:246]: It'd be nice to have numerics in here, too.

File: models/pattern/core/pattern.go [hunk 2]
```diff
@@ -202,23 +204,74 @@ func NewPatternFileFromCytoscapeJSJSON(byt []byte) (Pattern, error) {
...
+
+//processCytoElementsWithPattern iterates over all the cyto elements, convert each into a patternfile service and exposes a callback to handle that service
+func processCytoElementsWithPattern(eles []cytoscapejs.Element, pf *Pattern, callback func(svc Service, ele cytoscapejs.Element) error) error {
+	for _, elem := range eles {
 		// Try to create Service object from the elem.scratch's _data field
 		// if this fails then immediately fail the process and return an error
 		castedScratch, ok := elem.Scratch.(map[string]interface{})
 		if !ok {
-			return pf, fmt.Errorf("empty scratch field is not allowed, must containe \"_data\" field holding metadata")
+			return fmt.Errorf("empty scratch field is not allowed, must containe \"_data\" field holding metadata")
```
Comment (leecalcote) [inline models/pattern/core/pattern.go:263]:
```suggestion
			return fmt.Errorf("empty scratch field is not allowed, must contain \"_data\" field holding metadata")
```

**Comment by l5io** [comment]
Task linked: [CU-270rt0q Users unable to deploy designs due to service name requirements](https://app.clickup.com/t/270rt0q)

**Review by leecalcote — APPROVED** [review]

**Pull request merged**